<html><head><!-- This document was created for MSEQ2ALSC at Fri Jan 14 20:12:40 2005 from
../prosite/mseq2alsc.txt by MAN2HTML version 050114/2.0.6 -->
<title>Uppsala Software Factory - MSEQ2ALSC Manual</title>
<meta name="DESCRIPTION" content="Manual for Gerard 
Kleywegt's program MSEQ2ALSC">
<meta name="KEYWORDS" content="manual software Gerard 
Kleywegt program MSEQ2ALSC structural biology 
protein X-ray crystallography macromolecular molecular
Uppsala Software Factory bioinformatics">
</head><body bgcolor="#fefefe" background="back.gif">

<A HREF="http://xray.bmc.uu.se/usf/" target="_top">
<IMG ALIGN="right" BORDER="0" SRC="usf_logo.gif"
ALT="Uppsala Software Factory"></A>

<H2>  Uppsala Software Factory - MSEQ2ALSC Manual</H2> <P>   
<UL>  
<LI><H3>  <A HREF="#H1">1  </A>MSEQ2ALSC - GENERAL INFORMATION</H3> 
<LI><H3>  <A HREF="#H2">2  </A>REFERENCES</H3> 
<LI><H3>  <A HREF="#H3">3  </A>VERSION HISTORY</H3> 
<LI><H3>  <A HREF="#H4">4  </A>DESCRIPTION</H3> 
<LI><H3>  <A HREF="#H5">5  </A>USE WITH STRUPRO</H3> 
<LI><H3>  <A HREF="#H6">6  </A>USE WITH PRF2MSEQ</H3> 
<LI><H3>  <A HREF="#H7">7  </A>KNOWN BUGS</H3> 
<LI><H3>  <A HREF="#H8">8  </A>UNKNOWN BUGS</H3> 
</UL> 
<P>   
<HR NOSHADE>   <CENTER><H2>  <A NAME="H1">1 </A>MSEQ2ALSC - GENERAL INFORMATION</H2> </CENTER>
<P>   
Program : MSEQ2ALSC
 <BR> Version : 980212
 <BR> Author : Gerard J. Kleywegt, Dept. of Cell and Molecular Biology,
Uppsala University, Biomedical Centre, Box 596,
SE-751 24 Uppsala, SWEDEN
 <BR> E-mail : gerard@xray.bmc.uu.se
 <BR> Purpose : convert multiple sequence alignment to ALSCRIPT format
 <BR> Package : SBIN
<P>   
<HR NOSHADE>   <CENTER><H2>  <A NAME="H2">2 </A>REFERENCES</H2> </CENTER>
<P>   
Reference(s) for this program:
<P>   
* 1 * G.J. Kleywegt &amp; T.A. Jones (1998). Databases in protein
crystallography. Acta Cryst D54, 1119-1131.
[http://xray.bmc.uu.se/gerard/papers/databases.html]
[http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?db=PubMed&amp;cmd=Retrieve&amp;list_uids=10089488&amp;dopt=Citation]
[http://scripts.iucr.org/cgi-bin/paper?ba0001]
<P>   
* 2 * Kleywegt, G.J., Zou, J.Y., Kjeldgaard, M. &amp; Jones, T.A. (2001).
Around O. In: &quot;International Tables for Crystallography, Vol. F.
Crystallography of Biological Macromolecules&quot; (Rossmann, M.G.
&amp; Arnold, E., Editors). Chapter 17.1, pp. 353-356, 366-367.
Dordrecht: Kluwer Academic Publishers, The Netherlands.
<P>   
<HR NOSHADE>   <CENTER><H2>  <A NAME="H3">3 </A>VERSION HISTORY</H2> </CENTER>
<P>   
980211 - 0.1 - first version
 <BR> 980212 - 1.0 - cleaned up code and manual
<P>   
<HR NOSHADE>   <CENTER><H2>  <A NAME="H4">4 </A>DESCRIPTION</H2> </CENTER>
<P>   
MSEQ2ALSC is a simple non-interactive program that reads a
multiple-sequence alignment file as produced by STRUPRO
or PRF2MSEQ, and outputs a file that can be used as
input to Geoff Barton's ALSCRIPT program (see his
<A HREF="http://barton.ebi.ac.uk/">Web-site</A>).
<P>   
Usage: MSEQ2ALSC &lt; mseq_file &gt; alsc_file
<P>   
<HR NOSHADE>   <CENTER><H2>  <A NAME="H5">5 </A>USE WITH STRUPRO</H2> </CENTER>
<P>   
The multiple-sequence alignment file produced by STRUPRO,
together with MSEQ2ALSC and ALSCRIPT, enables you to
produce cute structure-based sequence alignments. MSEQ2ALSC
will clearly flag parts of the sequences that are and those
that are not structurally aligned. The latter will be
enclosed by &quot; &lt;&quot; and &quot;&gt; &quot; in the output file.
<P>   
HINT: use reasonable distance cut-offs in STRUPRO (e.g., 3.5 A
equivalent, and 5.0 A extension, or even 2.5 and 3.5 A) if
you intend to use the sequence alignment with ALSCRIPT to
prevent erroneous alignments.
<P>   
HINT: if you align multiple domain proteins, treat the individual
domains separately in STRUPRO and MSEQ2ALSC. Afterwards, merge
the two ALSCRIPT input files.
<P>   
<PRE> <TT>  <B>   
 ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE -----
 ../MSEQ2ALSC &lt; aligned.mseq &gt; q.alsc
 ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE -----
</B>  </TT> </PRE>
<P>   
Part of the input file:
<P>   
<PRE> <TT>  <B>   
 ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE -----
!
! Sequence alignment file
! Created by STRUPRO V. 980211/1.2 at Thu Feb 12 13:28:44 1998 for user gerard
!
<P>   
! NOT ALIGNED MOL   1 FROM PRO-   1 TO PRO-   1
!&gt; P
! NOT ALIGNED MOL   2 FROM CYS-   1 TO ASP-   2
!&gt; CD
! NOT ALIGNED MOL   3 FROM CYS-   1 TO ASP-   2
!&gt; CD
! NOT ALIGNED MOL   4 FROM PRO-   1 TO PRO-   1
!&gt; P
 ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE -----
</B>  </TT> </PRE>
<P>   
Part of the output file:
<P>   
<PRE> <TT>  <B>   
 ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE -----
&gt;MSEQ2ALSC_1
&gt;MSEQ2ALSC_2
&gt;MSEQ2ALSC_3
&gt;MSEQ2ALSC_4
<P>   
[...]
<P>   
&gt;MSEQ2ALSC_24
*
<P>   
&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;
PCCPPPPVVVVV----CCCCCTTS
-DD---VKDDDD----DDDDDKKN
-------E----------------
&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;
<P>   
NAANNNDFAAAAAAAAAAAAADDK
FFFFFFFAFFFFFFFFFFFFFQQF
SVVASANGLLLLDDDDVVVVVNNL
GGGGGGGIGGGGGGGGGGGGGGGG
NTTTNTYKTTTTTTTTTTTTTTTT
WWWWWWWYWWWWWWWWWWWWWWWW
KKKKKKKKKKKKKKKKKKKKKEEK
<P>   
[...]
<P>   
VEEVVVKKEEEEKKKKEEEEEKKE
RRRRRRKAKKKKKKKKRRRRRKKK
EAAEEEVQEEEEEEEEAAAAAKKV
<P>   
&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;
------H-----------------
&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;
<P>   
*
 ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE -----
</B>  </TT> </PRE>
<P>   
<HR NOSHADE>   <CENTER><H2>  <A NAME="H6">6 </A>USE WITH PRF2MSEQ</H2> </CENTER>
<P>   
If you use a multiple, partial sequence alignment file produced
by PRF2MSEQ, you will get an alignment of the residues that
matched the profile. Parts in between profile fragments will
be indicated by &quot; &lt;...&gt; &quot; in the output file.
<P>   
<PRE> <TT>  <B>   
 ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE -----
 ../MSEQ2ALSC &lt; prf.mseq &gt; q.alsc
 ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE -----
</B>  </TT> </PRE>
<P>   
Part of the input file:
<P>   
<PRE> <TT>  <B>   
 ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE -----
!    491 pos.    127 -   205 P37366|CCL1_YEAST CYCLIN CCL1.
          IAQHLNL PTEVVATAISFFRRFFLENS DPKSIVHTTIFLACKSENYFI DSFAQKAKSTRDS      VLKFEFKLLESL
!    512 pos.    209 -   296 P46277|CG1B_MEDVA G2/MITOTIC-SPECIFIC CYCLIN 1 (B-LIKE CYCLIN).
    VDWLIEVHDKFDL MHETLFLTVNLIDRFLEKQS KLQLVGLVAMLLACKYEEVSV    PVVGDLILIS YTRKEVLEMEKVMVNAL
!    402 pos.     53 -   146 P55168|CG1C_CHICK G1/S-SPECIFIC CYCLIN C.
            EHLKL RQQVIATATVYFKRFYARYS DPVLMAPTCVFLASKVEEFGV ISAATSVLKTRFS YRMNHILECEFYLLELM
!    356 pos.    175 -   238 P55168|CG1C_CHICK G1/S-SPECIFIC CYCLIN C.
                        LAWRIVNDTYRTDL PPFMIALACLHVACVVQQKDA             R VDMEKILEIIRVILKLY
!    444 pos.     53 -   146 P25008|CG1C_DROME G1/S-SPECIFIC CYCLIN C.
            EQLKL RQQVIATATVYFKRFYARNS DPLLLAPTCILLASKVEEFGV ISNSRLISICQSA YRTNHILECEFYLLENL
 ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE -----
</B>  </TT> </PRE>
<P>   
Part of the output file:
<P>   
<PRE> <TT>  <B>   
 ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE -----
&gt;MSEQ2ALSC_1
&gt;MSEQ2ALSC_2
&gt;MSEQ2ALSC_3
<P>   
[...]
<P>   
&gt;MSEQ2ALSC_90
*
-----------------------------I-------V-V-V------------------------------------------------
-----------------------------N-------N-N-N------------------------------------------------
-----------------------------Q-------Q-Q-Q----------------------------------------------R-
-----------------------------F-------F-F-F----------------------------------------------M-
<P>   
[...]
<P>   
DIF-ST-T-GGGGGGNGVVLGIGLGGNNDGIDGDNDDGDGDGDDIIIITDI-IIINYNIIIDYDVVVVVVNNNNDPNNNNEENNNNEH-G
SSSRARRSRAAAAAATSAVSASASAGCTTYTTYTTSTYTYTYTTATTTSTT-TAAASASSSTTTTTTTTTAAAAASSSSSKDSSSSEA-A
<P>   
&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;
..........................................................................................
..........................................................................................
&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;
<P>   
-YYVYYVYVCCCCCCYYFFYCYFYCYYYY-YY-YYYY-Y-Y-YYYYYYCYY-YYYYYYFFFYYYYYYYYYYYYYYFIIIIYFIIVVFV-A
-TRDRRDRDTTTSSTSTSTTSSTTTDSTTTRNTTSTTTSTSTETSTSSTTS-TTTTSTTTTTTSTTTTTTNTTTKERRRRTEKKKKES-S
-RMMTMMMMEEEGGEDPDVHVNVHEERRKLVKLAQKKLKLKLQSHKKKHAK-KRRKEKVVVKLKKKKKKKSSSARAPPPPDAPPPPAP-K
-KNENNENEDDRDDDKEKDEEEEEDENDKEPKEKQAKEKEKEKQKAKEARE-AKKAKKSSSKEKHHHHHAASSSEKDEEENKQQQQKR-N
-EHKHHKHKDEDEEEQEEEHDQDQEEDDQTDQHQEQQSQSQSQQQQQKEQK-QDDDEEDDDQTQQQQQQQEQQQETEEEESTEEEETQ-D
VVIIIIIIIIIIIIIIVIIIIVVIIILIVLIVIVVVVLVLVLVVIIIIVII-IIIIIILLLVLLIIIIIIVIIILILLLLIILLLLIL-I
LLLLLLLLLLLLLLTLVLILKLCLKLLILKLLKLVLLKLKLKLLLRLRIRR-RLLRYLMMMLKLRRRRRRRRRRIKLLLLRKLLLLQR-K
KEEEEEEEESSNTTRVQDRTERVAEQDRRPLRPRKRRPRPRPRRVQAQDQQ-SEETPERRRKPRQQQQQDEEEEARQQQQPREEEERD-N
FMCICCICIMMHMMMMAAAMGMAMGAAAMCMMCMMMMCMCMCMMMMMMMMM-MMMMVMMMMMCMMMMMMMMMMMMMMMMMDM----MW-A
EEEIEEIEIEEEEEEEEEEEEEEEEEEEELEELEEEELELELEEEEEEEEE-EEEEMEEEEEIEEEEEEEEEEEEEEEEEEE----EE-E
FKFRFFRFRILKLLLKKRRKRKRKKRRQHLRHMHAQHMHMHVHHKIQQRLQ-CKKQQQKKKHLHMMMMMMITTMTLLLLLLL----LV-M
KVYVYYVYVIIIMMIKYHYTFKYTFYTYLDFLDLDVLDLDLDLLTATKQQK-NLLTKHIIILDVKKKKKKTLLISLLLLLLL----LL-F
LMLILLILIIILIMIIMIMIIIMIIIIMVLLILILIVLVLVLIIIMIMIIM-IMMILIVVVVLLIIIIIIIIIIIVLLLLIV----IV-M
LVLLLLLLLMMLMMMLLLLLLLLLLLLILHLLHLLLLHLHLHLLLLLLLFL-LLLLALLLLLYLLLLLLLLLLLLLVVVVMV----LL-L
ENEKEEKEKKKQKKKGTA-NEGNNKRNDKQ-KRKKKKQKQKQKKSKNKSKK-RNNKQKEEEKQKRRRRRRKKKRRSNNNNET----S--T
SALLNLLLLEAAAA-AIT-KKAVKTV-TVT-VTTTIVTVTVTVVT-SATAT-RTTTLK---VTVVAVVVVEEELTVKKKK-T----T--S
LLMYLMYMYLLLLL-LLL-LLLLLLL-LLY-LYLLLLYLYLYLLL-L-LI--LLLLVL---LYLLLLLLLLLLLLLLLLL-L----L--L
*
 ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE -----
</B>  </TT> </PRE>
<P>   
<HR NOSHADE>   <CENTER><H2>  <A NAME="H7">7 </A>KNOWN BUGS</H2> </CENTER>
<P>   
None, at present (&quot;peppar, peppar&quot;).
<P>   
<HR NOSHADE>   <CENTER><H2>  <A NAME="H8">8 </A>UNKNOWN BUGS</H2> </CENTER>
<P>   
Does not compute.
<HR NOSHADE>    <A HREF="http://xray.bmc.uu.se/usf/" target="_top">
<IMG ALIGN="right" BORDER="0" SRC="usf_logo.gif"
ALT="Uppsala Software Factory"></A>
<FONT SIZE=-1>
<B> <I> Created at Fri Jan 14 20:12:40 2005
by MAN2HTML version 050114/2.0.6 . This manual
describes MSEQ2ALSC, a program of the <A HREF="http://xray.bmc.uu.se/usf/" target="_top">Uppsala
Software Factory</A> (USF), written and maintained by <A HREF="http://xray.bmc.uu.se/gerard"
target="_blank">Gerard Kleywegt</A>. &copy; 1992-2005.
</I> </B> </FONT><P>
<IMG SRC="http://xray.bmc.uu.se/cgi-bin/nph-count?width=6&link=http://alpha2.bmc.uu.se/~gerard/www/manuals/mseq2alsc_man.html">
</body></html>
