<html><head><!-- This document was created for STRUPAT at Fri Jan 14 20:12:42 2005 from
../prosite/strupat.txt by MAN2HTML version 050114/2.0.6 -->
<title>Uppsala Software Factory - STRUPAT Manual</title>
<meta name="DESCRIPTION" content="Manual for Gerard 
Kleywegt's program STRUPAT">
<meta name="KEYWORDS" content="manual software Gerard 
Kleywegt program STRUPAT structural biology 
protein X-ray crystallography macromolecular molecular
Uppsala Software Factory bioinformatics">
</head><body bgcolor="#fefefe" background="back.gif">

<A HREF="http://xray.bmc.uu.se/usf/" target="_top">
<IMG ALIGN="right" BORDER="0" SRC="usf_logo.gif"
ALT="Uppsala Software Factory"></A>

<H2>  Uppsala Software Factory - STRUPAT Manual</H2> <P>   
<UL>  
<LI><H3>  <A HREF="#H1">1  </A>STRUPAT - GENERAL INFORMATION</H3> 
<LI><H3>  <A HREF="#H2">2  </A>REFERENCES</H3> 
<LI><H3>  <A HREF="#H3">3  </A>VERSION HISTORY</H3> 
<LI><H3>  <A HREF="#H4">4  </A>INTRODUCTION</H3> 
<LI><H3>  <A HREF="#H5">5  </A>INPUT TO THE PROGRAM</H3> 
<H5><A HREF="#S1">5.1  </A>Start-up</H5>
<H5><A HREF="#S2">5.2  </A>Random-number seed</H5>
<H5><A HREF="#S3">5.3  </A>Random sequence</H5>
<H5><A HREF="#S4">5.4  </A>Cut-off distances and frameshifts</H5>
<H5><A HREF="#S5">5.5  </A>Minimum pattern length</H5>
<H5><A HREF="#S6">5.6  </A>Substitution-group mode</H5>
<H5><A HREF="#S7">5.7  </A>Little variation</H5>
<H5><A HREF="#S8">5.8  </A>PDB file</H5>
<LI><H3>  <A HREF="#H6">6  </A>OUTPUT</H3> 
<LI><H3>  <A HREF="#H7">7  </A>RESULTS</H3> 
<LI><H3>  <A HREF="#H8">8  </A>PATTERN REDUCTION</H3> 
<LI><H3>  <A HREF="#H9">9  </A>KNOWN BUGS</H3> 
<LI><H3>  <A HREF="#H10">10  </A>UNKNOWN BUGS</H3> 
</UL> 
<P>   
<HR NOSHADE>   <CENTER><H2>  <A NAME="H1">1 </A>STRUPAT - GENERAL INFORMATION</H2> </CENTER>
<P>   
Program : STRUPAT
 <BR> Version : 041001
 <BR> Author : Gerard J. Kleywegt, Dept. of Cell and Molecular Biology,
Uppsala University, Biomedical Centre, Box 596,
SE-751 24 Uppsala, SWEDEN
 <BR> E-mail : gerard@xray.bmc.uu.se
 <BR> Purpose : generate PROSITE patterns from aligned 3D protein structures
 <BR> Package : SBIN
<P>   
<HR NOSHADE>   <CENTER><H2>  <A NAME="H2">2 </A>REFERENCES</H2> </CENTER>
<P>   
Reference(s) for this program:
<P>   
* 1 * G.J. Kleywegt &amp; T.A. Jones (1998). Databases in protein
crystallography. Acta Cryst D54, 1119-1131.
[http://xray.bmc.uu.se/gerard/papers/databases.html]
[http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?db=PubMed&amp;cmd=Retrieve&amp;list_uids=10089488&amp;dopt=Citation]
[http://scripts.iucr.org/cgi-bin/paper?ba0001]
<P>   
* 2 * Kleywegt, G.J., Zou, J.Y., Kjeldgaard, M. &amp; Jones, T.A. (2001).
Around O. In: &quot;International Tables for Crystallography, Vol. F.
Crystallography of Biological Macromolecules&quot; (Rossmann, M.G.
&amp; Arnold, E., Editors). Chapter 17.1, pp. 353-356, 366-367.
Dordrecht: Kluwer Academic Publishers, The Netherlands.
<P>   
<HR NOSHADE>   <CENTER><H2>  <A NAME="H3">3 </A>VERSION HISTORY</H2> </CENTER>
<P>   
970512 - 0.1 - first version
 <BR> 970804 - 0.5 - first documented version
 <BR> 970805 - 0.6 - try to extend alignments backwards as well; minor changes
 <BR> 971030 - 1.0 - cleaned up code and manual
 <BR> 010227 - 1.1 - calculated empirical estimate for the probability of
each pattern using the formula used in EMOTIF
(CG Nevill-Manning, TD Wu, DL Brutlag, PNAS 95,
5865-5871 (1998)); also implemented EMOTIF's
substitution-group mode and made it the default
 <BR> 020823 - 1.2 - skip alt. conf. (B, C, ...) when reading PDB files
 <BR> 041001 - 1.3 - replaced Kabsch' routine U3BEST by quaternion-based
routine (U3QION) to do least-squares superpositioning
<P>   
<HR NOSHADE>   <CENTER><H2>  <A NAME="H4">4 </A>INTRODUCTION</H2> </CENTER>
<P>   
This program generates PROSITE patterns from a set of aligned
three-dimensional protein structures in PDB format.
<P>   
Suppose that you solve a new protein structure which turns out
to contain a fold which is (partly) similar to that of one or
more other proteins (e.g., using DEJAVU or SPASM). If you
align the two structures (e.g., using LSQMAN), you can feed
them into the program STRUPAT which will look for more or
less conserved residues in structurally conserved regions.
It will use these to generate PROSITE-type sequence patterns
(a.k.a. footprints, fingerprints, motifs, ...).
<P>   
Such a pattern may look as follows: G-x(3)-C-x(2)-[ILV]. This
means: glycine - three residues of any type - cysteine -
two residues of any type - one residue of type Ile/Leu/Val.
A protein which contains the peptide GYAVCPSV would fit this
pattern.
<P>   
If you want to scan PROSITE (
<A HREF="http://www.expasy.ch/sprot/prosite.html" target="_blank">
http://www.expasy.ch/sprot/prosite.html</A>
) patterns against the SWISS-PROT (and TREMBL) database, you
can use the WWW-based PROSITE server (
<A HREF="http://www.expasy.ch/sprot/scnpsit2.html" target="_blank">
http://www.expasy.ch/sprot/scnpsit2.html</A>
) at ExPASy in Geneva.
<P>   
<HR NOSHADE>   <CENTER><H2>  <A NAME="H5">5 </A>INPUT TO THE PROGRAM</H2> </CENTER>
<P>   
<HR NOSHADE>   <H3>  <A NAME="S1">5.1 </A>Start-up</H3> 
<P>   
When you start the program, it prints some information:
<P>   
<PRE> <TT>  <B>   
 ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE -----
 *** STRUPAT *** STRUPAT *** STRUPAT *** STRUPAT *** STRUPAT *** STRUPAT ***
<P>   
 Version  - 010227/1.1
 (C) 1992-2001 Gerard J. Kleywegt, Dept. Cell Mol. Biol., Uppsala (S)
 User I/O - routines courtesy of Rolf Boelens, Univ. of Utrecht (NL)
 Others   - T.A. Jones, G. Bricogne, Rams, W.A. Hendrickson
 Others   - W. Kabsch, CCP4, PROTEIN, E. Dodson, etc. etc.
<P>   
 Started  - Wed Feb 28 01:50:07 2001
 User     - gerard
 Mode     - interactive
 Host     - sarek
 ProcID   - 14618
 Tty      - /dev/ttyq14
<P>   
 *** STRUPAT *** STRUPAT *** STRUPAT *** STRUPAT *** STRUPAT *** STRUPAT ***
<P>   
 Reference(s) for this program:
<P>   
 *  1 * G.J. Kleywegt &amp; T.A. Jones (1998). Databases in protein
        crystallography. Acta Cryst D54, 1119-1131.
        [http://xray.bmc.uu.se/gerard/papers/databases.html]
        [http://www.ncbi.nlm.nih.gov/htbin-post/Entrez/query?uid=10089488&amp;form=6&amp;db=m&amp;Dopt=b]
        [http://www.iucr.org/iucr-top/journals/acta/tocs/actad/1998/actad5406_1.html]
<P>   
 *  2 * G.J. Kleywegt, J.Y. Zou, M. Kjeldgaard &amp; T.A. Jones (2001).
        Chapter 17.1.  Around O. Int. Tables for Crystallography,
        Vol. F. (In press.)
<P>   
 ==&gt; For manuals and up-to-date references, visit:
 ==&gt;     http://xray.bmc.uu.se/usf
 ==&gt; For downloading up-to-date versions, visit:
 ==&gt;     ftp://xray.bmc.uu.se/pub/gerard
<P>   
 *** STRUPAT *** STRUPAT *** STRUPAT *** STRUPAT *** STRUPAT *** STRUPAT ***
<P>   
 Max nr of atoms/residues       : (      50000)
 Max nr of molecules            : (        500)
 Max nr of residues in sequence : (       2000)
 Max nr of PROSITE patterns     : (        500)
 Random sequence length         : (    2000000)
 One-letter codes   : ( A R N D C E Q G H I L K M F P S T W Y V)
 Three-letter codes : ( ALA ARG ASN ASP CYS GLU GLN GLY HIS ILE LEU LYS
  MET PHE PRO SER THR TRP TYR VAL)
 ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE -----
</B>  </TT> </PRE>
<P>   
<HR NOSHADE>   <H3>  <A NAME="S2">5.2 </A>Random-number seed</H3> 
<P>   
The first bit of input is an integer seed for the random-number
generator. This will be used to generate a random amino-acid
sequence. If you repeat this run of the program on the same
machine with the same seed, you should be getting identical
results.
<P>   
<PRE> <TT>  <B>   
 ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE -----
 Random-number seed ? (  123456)
 Random-number seed : (  123456)
 =&gt; Random number generator initialised with seed :     123456
 ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE -----
</B>  </TT> </PRE>
<P>   
<HR NOSHADE>   <H3>  <A NAME="S3">5.3 </A>Random sequence</H3> 
<P>   
The program will now generate a random amino-acid sequence of (at
present) 2,000,000 residues. This sequence has an aminoa-acid
distribution similar to that found in proteins in the PDB (GJK,
unpublished results). It will be used later to test how often
generated PROSITE patterns occur in this sequence, which gives
you some idea of the pattern occurring by chance. Of course,
a random sequence is unlikely to be &quot;protein-like&quot;, but if a
pattern matches the random sequence more than, say, 5 or 10 times,
it is unlikely to be a very discriminating one.
<P>   
<PRE> <TT>  <B>   
 ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE -----
 Generating random sequence ...
 Target composition    : (   0.081    0.044    0.046    0.058    0.019
  0.058    0.037    0.080    0.022    0.053    0.081    0.059    0.020
  0.040    0.047    0.068    0.063    0.016    0.038    0.071)
 Working ...
 Actual composition    : (   0.081    0.044    0.046    0.058    0.019
  0.057    0.037    0.080    0.022    0.053    0.081    0.060    0.020
  0.040    0.046    0.068    0.063    0.015    0.038    0.070)
 ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE -----
</B>  </TT> </PRE>
<P>   
<HR NOSHADE>   <H3>  <A NAME="S4">5.4 </A>Cut-off distances and frameshifts</H3> 
<P>   
You are to provide a cut-off distance (in &Aring;) for CA atoms of
different molecules to be considered equivalent. If this
number is very high, frameshifts may occur in the structural
alignments, although the program can be instructed to try
and correct for these. Another cut-off distance determines
how bits of equivalent structure are extended at their ends.
<P>   
<PRE> <TT>  <B>   
 ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE -----
 Equivalent CA distance ? (   3.500) 5
 Equivalent CA distance : (   5.000)
<P>   
 Extension CA distance ? (   5.000) 8
 Extension CA distance : (   8.000)
<P>   
 Try to correct frame-shifts (Y/N) ? (Y)
 Try to correct frame-shifts (Y/N) : (Y)
 ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE -----
</B>  </TT> </PRE>
<P>   
<HR NOSHADE>   <H3>  <A NAME="S5">5.5 </A>Minimum pattern length</H3> 
<P>   
Very short patterns are unlikely to be very specific. Also, for
calculating RMSDs between aligned stretches, at least 3 CA atoms
are required.
<P>   
<PRE> <TT>  <B>   
 ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE -----
 Min pattern length ? (      10) 5
 Min pattern length : (       5)
 ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE -----
</B>  </TT> </PRE>
<P>   
<HR NOSHADE>   <H3>  <A NAME="S6">5.6 </A>Substitution-group mode</H3> 
<P>   
You can use either the old STRUPAT algorithm for grouping residues,
or the set that is used by EMOTIF (recommended).
<P>   
<PRE> <TT>  <B>   
 ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE -----
 Choose substitution-group mode:
   E = EMOTIF (recommended)
   S = STRUPAT
 Substitution-group mode (E/S) ? (E)
 Substitution-group mode (E/S) : (E)
 ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE -----
</B>  </TT> </PRE>
<P>   
<HR NOSHADE>   <H3>  <A NAME="S7">5.7 </A>Little variation</H3> 
<P>   
This block of input is only required when you use the old STRUPAT
substitution-group mode (not recommended !).
<P>   
If only 2, 3 or 4 different residue types occur in a certain position
of all structures/sequences, this can be included in the pattern.
But this only makes sense if you have a reasonable number of
structures. For instance, if you only have three structures,
and you observe residue types Arg, Lys, and Gln in a certain
position, you probably would not want to conclude that this
residue is always Arg, Lys or Gln. However, if you have 30 aligned
structures, you might.
<P>   
There are a few exceptions to this, namely if the various observed
residue types are similar, such as: D/E, R/K, F/Y, F/Y/W, N/Q,
S/T, A/G, and I/L/V.
<P>   
<PRE> <TT>  <B>   
 ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE -----
 If only 2, 3, or 4 different residue types
 occur in at least NMIN2, NMIN3, NMIN4 of
 your sequences, an entry will be generated
 (e.g., [SE], [TGW], [KILM]).  By setting
 NMIN2/3/4 greater than the number of sequences
 you can prevent that such entries are used.
 Value for MIN2 (&gt;2) ? (       6)
 Value for MIN3 (&gt;3) ? (      15)
 Value for MIN4 (&gt;4) ? (     100)
 Value for MIN2 : (       6)
 Value for MIN3 : (      15)
 Value for MIN4 : (     100)
 ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE -----
</B>  </TT> </PRE>
<P>   
<HR NOSHADE>   <H3>  <A NAME="S8">5.8 </A>PDB file</H3> 
<P>   
Provide the name of the PDB file which contains ALL molecules. Note
that the molecules must have been superimposed previously (e.g.,
with O or LSQMAN; LSQMAN contains a BRute_force command to find
structural alignments &quot;ab initio&quot;). Any two subsequent molecules
in the file must have different chain identifiers. However, not
all identifiers have to be unique (which would otherwise limit
you to a maximum of 26 molecules), e.g. you could alternate chain
identifiers A and B. Note that the program *ONLY* reads the CA
atoms, so you can make your files considerably smaller by only
including these (e.g.: grep ^ATOM myfile.pdb | grep ' CA ' &gt; new.pdb).
<P>   
The example below is for a PDB file which contains a number of
superimposed glutathione S-transferase structures.
<P>   
<PRE> <TT>  <B>   
 ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE -----
 Name of PDB file ? (aligned.pdb) 1rbp/aligned.pdb
 Name of PDB file : (1rbp/aligned.pdb)
<P>   
 Nr of CA atoms  : (        840)
 Nr of molecules : (          5)
<P>   
 Mol #   1 Atoms      1 to    174
 Mol #   2 Atoms    175 to    350
 Mol #   3 Atoms    351 to    510
 Mol #   4 Atoms    511 to    683
 Mol #   5 Atoms    684 to    840
 ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE -----
</B>  </TT> </PRE>
<P>   
<HR NOSHADE>   <CENTER><H2>  <A NAME="H6">6 </A>OUTPUT</H2> </CENTER>
<P>   
STRUPAT will now start looking for residues that are structurally
equivalent in all aligned structures (i.e., a residue in the first
protein has a partner in each of the other structures within the
cut-off distance). When it encounters such a residue, it checks to
see if neighbouring residues (on either side) also have partners
in all the other structures (now using the second distance cut-off).
<P>   
In this way, a set of residues is equivalenced between all structures.
However, the structural superposition may not always be optimal,
so the program will try to detect and fix any frameshift errors.
It does this simply by checking for each structure if shifting
the alignment to the first structure by one residue forward or
backward would improve the superpositioning RMSD. If so, the
equivalenced residues are altered accordingly, and the frameshift
test is carried out again, until no more frameshifts occur.
<P>   
<PRE> <TT>  <B>   
 ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE -----
 ----------------------------------------------------------------------
 Shift mol   5 by +1 (RMSD -1/0/+1 :    6.2   4.2   2.9 A)
<P>   
 ----------------------------------------------------------------------
 ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE -----
</B>  </TT> </PRE>
<P>   
At that stage, the program will again try to extend the alignments
in both directions using the extension distance cut-off. If the
resulting conserved set of residues contains at least the minimum
number of residues defined by the user, a potential pattern has
been found.
<P>   
For every (potential) pattern that the program discovers, the output
includes:
<P>   
- a listing of the first residue of the stretch of structurally
conserved residues in every molecule
<P>   
<PRE> <TT>  <B>   
 ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE -----
 New structurally conserved stretch !
 Starts at residue LYS -   12
   molecule    2 @ LYS -   12
   molecule    3 @ VAL -    3
   molecule    4 @ VAL -   15
   molecule    5 @ GLY -   11
 ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE -----
</B>  </TT> </PRE>
<P>   
- for every residue, the amino-acid type in each molecule, and the
program's &quot;reduction&quot; of it in terms of PROSITE pattern elements.
For instance, a strictly conserved glycine will be &quot;reduced&quot; to
&quot;G&quot;, whereas &quot;|YFFY|&quot; would yield &quot;[FY]&quot;. (The precise reduction
depends on the substitution-group mode chosen !)
<P>   
<PRE> <TT>  <B>   
 ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE -----
 |KKVVG| ==&gt;  X -
 |EEKDR| ==&gt;  X -
 |NNDNN| ==&gt;  [DN] -
 |FFFFF| ==&gt;  F -
 |DDDDN| ==&gt;  [DN] -
 |KKIWV| ==&gt;  X -
 |AASSE| ==&gt;  X -
 |RRKNK| ==&gt;  X -
 |FFFYI| ==&gt;  [FILVY] -
 |SALHN| ==&gt;  X -
 |GGGGG| ==&gt;  G -
 |TTFKE| ==&gt;  X -
 |WWWWW| ==&gt;  W -
 |YYYWH| ==&gt;  X -
 |AAEET| ==&gt;  X -
 |MMIVI| ==&gt;  [ILMV] -
 |AAAAI| ==&gt;  X -
 |KKFKL| ==&gt;  X -
 |KKAYA| ==&gt;  X -
 |DDSPS| ==&gt;  X -
 |PPKND| ==&gt;  X -
 |EEMSK| ==&gt;  X -
 ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE -----
</B>  </TT> </PRE>
<P>   
- some information about the aligned set of residues, namely their
number, and the RMS (RMSD) value (in &Aring;). This is calculated from
all Nmol*(Nmol-1)/2 possible pair-wise superpositionings of this
stretch of residues.
<P>   
<PRE> <TT>  <B>   
 ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE -----
 Nr of residues  : (      22)
 RMS (RMSD) (A)  : (   1.640)
 ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE -----
</B>  </TT> </PRE>
<P>   
- now the program goes to work and &quot;reduces&quot; the partial PROSITE
patterns, by collecting sequential &quot;X&quot;s and stripping any &quot;X&quot;s
from the start and end
<P>   
<PRE> <TT>  <B>   
 ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE -----
 PROSITE pattern : ([DN] - F - [DN] - X(3) - [FILVY] - X - G - X - W - X(2)
   - [ILMV])
 ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE -----
</B>  </TT> </PRE>
<P>   
- the program calculates the probability of finding this pattern
by chance (using the same formula as EMOTIF). The number of
false positives is roughly this probability times the number
of amino-acid residues in the database against which the scan
is performed (2,000,000 for STRUPAT's internal testing)
<P>   
<PRE> <TT>  <B>   
 ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE -----
 Probability     : (  3.711E-08)
 ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE -----
</B>  </TT> </PRE>
<P>   
- the program calculates two scores to help you judge the value
of the pattern.
<P>   
&quot;Score 1&quot; is calculated as SUM 10LOG(Ntotal/Nposs), where the sum
extends over all residues in the pattern, Ntotal is the number of
aligned structures, and Nposs the number of different residue
types that occur in each position IF (and only if) the residue
resulted in a non-&quot;X&quot; partial pattern. E.g., if there are four
different residue types for four sequences, (Ntotal/Nposs) will
be 1, and the contribution to the sum of logs will be zero. If
there are only two possible residue types observed in 30 different
structures, the contribution will be 10LOG(20/2), since the maximum
number of possible different residue types is 20. The higher the
total sum, the more specific information the pattern contains.
Usually, this is strongly correlated to the length of the pattern.
<P>   
&quot;Score 2&quot; is an integer number calculated as a sum over all residues
of the pattern of a subjective score of the quality of the
pattern element. The subjective score varies between 0 (for
an &quot;X&quot; entry) to 10 (for a strictly conserved residue type).
<P>   
<PRE> <TT>  <B>   
 ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE -----
 Score 1         : (   6.884)
 Score 2         : (      52)
 ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE -----
</B>  </TT> </PRE>
<P>   
- finally, the program will check the random sequence it prepared
earlier to see how often the pattern occurs in it. If it occurs
more than a few times, the pattern is probably not suitable
for searching againts a database, since it is likely to result
in many false positives.
<P>   
<PRE> <TT>  <B>   
 ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE -----
 Nr of matches to random sequence : (          0)
 ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE -----
</B>  </TT> </PRE>
<P>   
<HR NOSHADE>   <CENTER><H2>  <A NAME="H7">7 </A>RESULTS</H2> </CENTER>
<P>   
When the program has finished, it will print a summary of the
PROSITE patterns of sufficient length. For every pattern the
following is listed:
<P>   
- nr of the pattern
 <BR> - length (nr of residues that it spans)
 <BR> - probability of matching the pattern by chance
 <BR> - score
 <BR> - number of chance matches found in the program's random sequence
 <BR> - number of chance matches expected in the program's random sequence
<P>   
<PRE> <TT>  <B>   
 ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE -----
 Nr of PROSITE patterns found : (       3)
<P>   
    # Leng  Probability  Score      #Random    #Expected
    1   14   3.7111E-08    6.9            0   7.4223E-02
 [DN]-F-[DN]-X(3)-[FILVY]-X-G-X-W-X(2)-[ILMV]
<P>   
    # Leng  Probability  Score      #Random    #Expected
    2   11   3.8500E-06    5.3           10   7.7001E+00
 [IV]-X(2)-T-D-X(3)-[FY]-X-[ILMV]
<P>   
    # Leng  Probability  Score      #Random    #Expected
    3   33   3.1830E-05    2.6          217   6.3660E+01
 [ILV]-[FLY]-X-R-X(11)-[FILMV]-X(16)-[FILVY]
 ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE -----
</B>  </TT> </PRE>
<P>   
You can now take these patterns and scan them against SWISSPROT/TrEMBL.
<P>   
<HR NOSHADE>   <CENTER><H2>  <A NAME="H8">8 </A>PATTERN REDUCTION</H2> </CENTER>
<P>   
The program uses a simple algorithm to reduce a string of residue
types to a PROSITE sub-pattern. The original STRUPAT algorithm
uses the following cases:
<P>   
<PRE> <TT>  <B>   
 ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE -----
 CASE #     &quot;Score 2&quot;   Description
 ------     ---------   ---------------------------------------------------------
    1          10       absolutely conserved residue type
    2           8       (potentially) negatively charged residue [DE]
    3           8       (potentially) positively charged residue [RK]
    4           8       only Phe and Tyr occur
    5           6       only Phe and Tyr and Trp occur
    6           5       only Asn and Gln occur
    7           5       only Ser and Thr occur
    8           5       only Ala and Gly occur
    9           6       only Ile and Leu and Val occur
   10           3       only 2 different types occur and at least NMIN2 sequences
   11           2       only 3 different types occur and at least NMIN3 sequences
   12           1       only 4 different types occur and at least NMIN4 sequences
 ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE -----
</B>  </TT> </PRE>
<P>   
The recommended EMOTIF substitution-group mode uses the following
cases:
<P>   
<PRE> <TT>  <B>   
 ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE -----
 CASE #     &quot;Score 2&quot;   Description
 ------     ---------   ---------------------------------------------------------
    1          10       absolutely conserved residue type
    2           8       sets of 2: IV, FY, HY, KR, EQ, DE, DN, ST, AS
    3           6       sets of 3: ILV, FLY, FWY, KQR, EKQ, AST
    4           4       sets of 4: ILMV, EKQR
    5           2       sets of 5: FILMV, FILVY
    6           1       sets of 6: FILMVY
 ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE ----- EXAMPLE -----
</B>  </TT> </PRE>
<P>   
<HR NOSHADE>   <CENTER><H2>  <A NAME="H9">9 </A>KNOWN BUGS</H2> </CENTER>
<P>   
None, at present (&quot;peppar, peppar&quot;).
<P>   
<HR NOSHADE>   <CENTER><H2>  <A NAME="H10">10 </A>UNKNOWN BUGS</H2> </CENTER>
<P>   
Does not compute.
<HR NOSHADE>    <A HREF="http://xray.bmc.uu.se/usf/" target="_top">
<IMG ALIGN="right" BORDER="0" SRC="usf_logo.gif"
ALT="Uppsala Software Factory"></A>
<FONT SIZE=-1>
<B> <I> Created at Fri Jan 14 20:12:42 2005
by MAN2HTML version 050114/2.0.6 . This manual
describes STRUPAT, a program of the <A HREF="http://xray.bmc.uu.se/usf/" target="_top">Uppsala
Software Factory</A> (USF), written and maintained by <A HREF="http://xray.bmc.uu.se/gerard"
target="_blank">Gerard Kleywegt</A>. &copy; 1992-2005.
</I> </B> </FONT><P>
<IMG SRC="http://xray.bmc.uu.se/cgi-bin/nph-count?width=6&link=http://alpha2.bmc.uu.se/~gerard/www/manuals/strupat_man.html">
</body></html>
